### DISCUSSION TITLE
Net::ReadTimeout, Net::WriteTimeout, and SMTPSessionNotStartedError

### DISCUSSION THREAD

**Discussion by midlandss** [postalserver/postal#3082] [Help with using Postal]
I have a weird issue. One of my servers last Friday encountered an issue and attempted to send 150 messages via my postalserver.  About the same time, one of the staff attempted to send a newsletter out to all users (about 90) using an newsletter plugin on our website (hosted internally). However, something must have happened as almost half of the initial emails of 150 (75) made it through, but the remaining half stopped and all of the newsletters were held up. Delivery to recipients NOT hosted at Microsoft seem to be fine. It is any Microsoft 365 recipients are having issues. Occasionally, a single test email will make it through. I have tested SMTP on the postal server itself and I can successfully open a telnet connection and send test emails. Here are some of the issues/logs from postal:

August 04, 2024 15:31          Soft fail
Temporary SMTP delivery error when sending to 52.101.11.3:25 (midlandss-org.mail.protection.outlook.com)
Net::ReadTimeout with #<TCPSocket:(closed)>
Time: 60.3s
Support Ref: Q1AYIGXW

August 04, 2024 12:31           Soft fail
An error occurred while sending the message to 52.101.9.11:25 (midlandss-org.mail.protection.outlook.com)
SMTPClient::Endpoint::SMTPSessionNotStartedError
Time: 0.0s
Support Ref: XSN9BIKG

August 04, 2024 15:13            Soft fail
An error occurred while sending the message to 52.101.40.24:25 (campfirewtx-org.mail.protection.outlook.com)
Net::WriteTimeout with #<TCPSocket:(closed)>
Time: 60.27s
Support Ref: NZIHAUMQ

postal-worker  | 2024-08-04 20:45:05 +0000 INFO   queue latency is 153552.94303441048s component=worker thread=work0 original_queued_message=58536
postal-worker  | 2024-08-04 20:45:05 +0000 INFO   found 0 associated messages to process at the same time batch_key=outgoing-midlandss.org component=worker thread=work0 original_queued_message=58536
postal-worker  | 2024-08-04 20:45:06 +0000 INFO   Connected to 52.101.11.13:25 (midlandss-org.mail.protection.outlook.com) log_id=6THDN5P9 component=worker thread=work0 original_queued_message=58536 queued_message=58536
postal-worker  | 2024-08-04 20:45:06 +0000 INFO   Sending message 2::99087 to ghires@midlandss.org log_id=6THDN5P9 component=worker thread=work0 original_queued_message=58536 queued_message=58536
postal-worker  | 2024-08-04 20:45:43 +0000 DEBUG  acquired task role by renewing it component=worker thread=tasks
postal-worker  | 2024-08-04 20:45:43 +0000 INFO   running task component=worker thread=tasks task=PruneWebhookRequestsScheduledTask
postal-worker  | 2024-08-04 20:45:43 +0000 INFO   Pruning webhook requests for server 2 component=worker thread=tasks task=PruneWebhookRequestsScheduledTask
postal-worker  | 2024-08-04 20:45:43 +0000 INFO   scheduling task to next run at 2024-08-04 21:45:00 UTC component=worker thread=tasks task=PruneWebhookRequestsScheduledTask
postal-worker  | 2024-08-04 20:45:43 +0000 INFO   running task component=worker thread=tasks task=SendNotificationsScheduledTask
postal-worker  | 2024-08-04 20:45:43 +0000 INFO   scheduling task to next run at 2024-08-04 20:46:43 UTC component=worker thread=tasks task=SendNotificationsScheduledTask
postal-worker  | 2024-08-04 20:45:43 +0000 INFO   running task component=worker thread=tasks task=TidyQueuedMessagesTask
postal-worker  | 2024-08-04 20:45:43 +0000 INFO   scheduling task to next run at 2024-08-04 21:45:00 UTC component=worker thread=tasks task=TidyQueuedMessagesTask
postal-worker  | 2024-08-04 20:46:06 +0000 ERROR  Net::WriteTimeout: Net::WriteTimeout with #<TCPSocket:fd 7, AF_INET, 10.10.x.x, 59858> log_id=6THDN5P9 component=worker thread=work0 original_queued_message=58536 queued_message=58536
postal-worker  | 2024-08-04 20:46:06 +0000 INFO   message requeued for trying later retry_after=2024-08-04 23:17:33 UTC component=worker thread=work0 original_queued_message=58536 queued_message=58536

Any ideas how I should proceed to troubleshoot?  I've had to modify the SMTP settings of a ton of servers/services to utilize another 3rd party SMTP system so notifications, alerts, voicemails, etc. can be delivered until I can figure things out. Any ideas or assistance would greatly be appreciated!

**Comment by willpower232** [comment]
> Delivery to recipients NOT hosted at Microsoft seem to be fine. It is any Microsoft 365 recipients are having issues. Occasionally, a single test email will make it through.

I would presume you have somehow upset one of their limits or firewalls, it is also possible that someone from a "nearby" IP address has been even more annoying to them and they just blocked the whole range of IP addresses.

**Comment by midlandss** [comment]
Thank you for responding, Will. I greatly appreciate your advice and expertise!

However, if Microsoft was blocking my IP address, why is it only blocking when postal is trying to connect?  I can perform a manual telnet session using the exact IP/fqdn that is indicated as having issues with delivery in the postal logs and successfully connect and send a test message using the same to/from address and it goes through?  I completed a "delist" request on Microsoft's sender.office.com and they indicated my IPs were not being throttled or blocked.

UPDATE 5/8/2024, 2:00 PM CDT
I'm receiving the same timeout issues with emails to Google as well:
August 05, 2024 13:57                      Soft fail
Temporary SMTP delivery error when sending to 74.125.137.27:25 (ASPMX.L.GOOGLE.com)
Net::ReadTimeout with #<TCPSocket:(closed)>
Time: 60.64s
Support Ref: LUOZP0IX

Can you help me understand the difference between these:
Net::ReadTimeout with #<TCPSocket:(closed)>
Net::WriteTimeout with #<TCPSocket:(closed)>
SMTPClient::Endpoint::SMTPSessionNotStartedError

Is it possible there is some mechanism or issue with postal?  Any ideas how I might dig deeper or troubleshoot?

![image](https://github.com/user-attachments/assets/2633d4d4-72bd-4531-b9ca-71daca116ffb)

**Reply by willpower232** [reply]
Is it possible that your server provider is blocking your outgoing traffic somehow?
